component/componenttest/error_waiting_host.go

```diff
@@ -12,38 +12,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package extensiontest
+package componenttest
 
 import (
 	"time"
 
 	"github.com/open-telemetry/opentelemetry-collector/component"
 )
 
-// MockHost mocks an component.Host for test purposes.
-type MockHost struct {
+// ErrorWaitingHost mocks an component.Host for test purposes.
+type ErrorWaitingHost struct {
```

Comment:
Nit: instead of NoopHost and ErrorWaitingHost we could probably just eliminate 2 mock hosts and keep only one MockHost that does both.
Up to you.

Reply:
I am worried about the number of items to set to the channel, should that be `100` or something like that for the case where we do test for errors and the implementation reports a fatal error to the host? I think about a large number because we don't want to block if people don't want to check for errors. What do you think?